docs/DevGuide/WritingDox.md

```diff
@@ -31,11 +31,75 @@ While the `file_name` portion is not necessary, it is useful for reducing the
 likelihood of reference collisions. You can add a table of contents using the
 Doxygen command <code>\\tableofcontents</code>. All sections, subsections,
 subsubsections, etc. are shown in the table of contents if they have a tag,
-if not they do not appear. 
+if not they do not appear.
 
 # C++ Documentation {#writing_dox_cpp_dox_help}
 
-Doxygen comments...
+We encourage you to add Doxygen documentation for any of the following
+you may have added in a new hpp file:
+
++ classes
+
++ structs
+
++ enums
+
++ functions
+
++ namespaces
+
+Documentation begins on the line immediately above the declaration with
+either a triple slash `///` or a `/*!`. Build your documentation by running
```

Comment:
may want to provide examples of block comments illustrating the `///` vs `/*!`, ` * `, ` */` syntax